src/CgiHandler.cpp

```diff
@@ -1,51 +1,70 @@
 
 #include "CgiHandler.hpp"
 
-static std::map<std::string, std::string> initCgiEnvironment(HttpRequest request)
+static const char *transferToString(std::map<std::string, std::string> input)
 {
-	std::map<std::string, std::string> cgiEnvironment;
+	std::string base = "";
 
-	cgiEnvironment["REQUEST_URI"] = request.getUri();
-	cgiEnvironment["GATEWAY_INTERFACE"] = GATEWAY_VERSION;
-	cgiEnvironment["SERVER_PROTOCOL"] = HTTP_VERSION;
-	cgiEnvironment["REQUEST_METHOD"] = request.translateMethod(request.getMethod());
-	cgiEnvironment["SCRIPT_FILENAME"] = FileHandler::getFilePath(request.getUri());
-	cgiEnvironment["SERVER_SOFTWARE"] = "SillyLittleSoftware/1.0";
-	cgiEnvironment["SERVER_NAME"] = "127.0.0.1";
-	
-	return cgiEnvironment;
+	for (std::map<std::string, std::string>::iterator it = input.begin(); it != input.end(); it++)
+	{
+		base.append(it->first);
+		base.append("=");
+		base.append(it->second);
+		base.append("&");
+	}
+	base.pop_back();
+	const char *string = base.c_str();
+	return string;
 }
 
-static char **transferToString(std::map<std::string, std::string> cgiEnvironment)
+
+static char **transferToStringArray(std::map<std::string, std::string> input)
 {
-	char **environmentString = NULL;
-	environmentString = new char*[cgiEnvironment.size() + 1]; 
+	char **stringArray = NULL;
+	stringArray = new char*[input.size() + 1];
 	int i = 0;
 
-	for (std::map<std::string, std::string>::iterator it = cgiEnvironment.begin(); it != cgiEnvironment.end(); ++it)
+	for (std::map<std::string, std::string>::iterator it = input.begin(); it != input.end(); it++)
 	{
 		std::string base = it->first;
 		base.append("=");
 		base.append(it->second);
-		environmentString[i] = strdup(base.c_str());
+		stringArray[i] = strdup(base.c_str());
 		i++;
 	}
-	environmentString[i] = nullptr;
+	stringArray[i] = nullptr;
 
-	return environmentString;
+	return stringArray;
 }
 
-std::string findShebang(std::string fullPath)
+
+static std::map<std::string, std::string> initCgiEnvironment(HttpRequest request)
+{
+	std::map<std::string, std::string> cgiEnvironment;
+
+	cgiEnvironment["REQUEST_URI"] = request.getUri();
+	cgiEnvironment["GATEWAY_INTERFACE"] = GATEWAY_VERSION;
+	cgiEnvironment["SERVER_PROTOCOL"] = HTTP_VERSION;
+	cgiEnvironment["REQUEST_METHOD"] = request.translateMethod(request.getMethod());
+	cgiEnvironment["SCRIPT_FILENAME"] = FileHandler::getFilePath(request.getUri());
+	cgiEnvironment["SERVER_SOFTWARE"] = "SillyLittleSoftware/1.0"; //fetch from config?
+	cgiEnvironment["SERVER_NAME"] = "127.0.0.1"; //fetch from config?
+	cgiEnvironment["QUERY_STRING"] = transferToString(request.getParameters());
+
+	return cgiEnvironment;
+}
+
+std::string findInterpreterPath(std::string fullPath)
```

Comment:
does this mean we could actually run other scripts than python too?

Reply:
I mean that this finds basically which command we are running aka given bash it would get that too? nice!

Reply:
in theory yes, if we add the other validation (and whatever config would need?) for the correct extension. Then we could just add the working scripts and make sure they do the response correctly.